### DISCUSSION TITLE
Foundry cast --to-base error

### DISCUSSION THREAD

**Discussion by mihirWeb** [Cyfrin/foundry-full-course-cu#1307] [Q&A]
I am calling a function with "cast call" cmd and it is returning the value in hex and to convert that hex into decimal i used "cast --to-base" cmd but it is showing this error and yes the output contains one string and one decimal number.

```
root@LAPTOP-FCMIQES2:/mnt/d/solidity/Foundry_demo/foundry-simple-storage# cast --to-base 0x0000000000000000000000000000000000000000000000000000000000000040000000000000000000000000000000000000000000000000000000000000030900000000000000000000000000000000000000000000000000000000000000056d69686972000000000000000000000000000000000000000000000000000000 dec
Error: 
The value is too large to fit the target type

Context:
- The value is too large to fit the target type
```

**Comment by xyz899** [comment]
Hi,

It's probably because your hexadecimal value is way too big for Foundry's CLI to handle, where did you get that from ? Y ou cast like this for example :

```
cast --to-base 0x0000000000000000000000000000000000000000000000000000000000000040000000000000000000000000000000000000000000000000000000000000030900000000000000000000000000000000000000000000000000000000000000056d69686972000000000000000000000000000000000000000000000000000000 dec
Error: 
The value is too large to fit the target type
Context:
- The value is too large to fit the target type

$ cast --to-base 0x68  dec
104
$ cast --to-base 0xb22  dec
2850 
```

Cast is used with max 107 characters with 0's ( 109 with 0x )

**Comment by mihirWeb** [comment]
yes I did the same way and it is showing error.
and I get this hex value as an output after calling a function from my smart contract i.e -

```
root@LAPTOP-FCMIQES2:/mnt/d/solidity/Foundry_demo/foundry-simple-storage# cast call 0x5FbDB2315678afecb367f032d93F642f64180aa3 "storageRetrive()" 
0x0000000000000000000000000000000000000000000000000000000000000040000000000000000000000000000000000000000000000000000000000000030900000000000000000000000000000000000000000000000000000000000000056d69686972000000000000000000000000000000000000000000000000000000
```

can you help me to fix this?

**Comment by xyz899** [comment]
This is encoded data. In Ethereum, when data (especially strings or arrays) is returned from a smart contract function, it's encoded in a specific way. So you can't just copy paste the output and get the specific number, it's an ABI-encoded return value. 

Use this for decoding :
https://book.getfoundry.sh/reference/cast/cast-calldata-decode

**Comment by mihirWeb** [comment]
Thanks a lot @Winter9998

**Reply by xyz899** [reply]
no problem select the answer so people know what to do after

**Comment by estebanzimanyi** [comment]
## Beta testing is now open — PR #113 ready for review

The implementation described in this RFC is now in a testable state on **MobilityDuck PR #13**: https://github.com/MobilityDB/MobilityDuck/pull/113

---

### For beta testers — try it in 15 minutes

Build from source (first build ~5-10 min; community extension coming once beta validates):

```bash
git clone --recurse-submodules https://github.com/MobilityDB/MobilityDuck.git
cd MobilityDuck
git checkout feat/edge-to-cloud-quickstart
make
```

Run the zero-data quickstart — no CSV needed:

```bash
TZ=UTC ./build/release/duckdb -c ".read examples/quickstart/quickstart.sql"
```

Expected: 5 vessels, geodetic lengths in metres (vessel 5 ≈ 172 km), vessels 1/2/4/5 inside the Copenhagen bounding box, vessel 3 absent.

Then run the same three queries on MobilityDB to verify cross-platform portability:

```bash
psql -d <your_db> -f examples/quickstart/quickstart_mobilitydb.sql
```

To try with your own GPS CSV: edit the five `CONFIGURE` macros in `examples/generic-ingest/generic_ingest.sql` and run it.

Full guide: [`docs/beta-testing-edge-to-cloud.md`](https://github.com/MobilityDB/MobilityDuck/blob/feat/edge-to-cloud-quickstart/docs/beta-testing-edge-to-cloud.md)

---

### For MobilityDB committers

The guide above has a **Part 2** specifically for committers: PR #13 links, implementation status table, geodetic fix detail (`geom_to_geog()` replacing `FLAGS_SET_GEODETIC`), and review checklist.

Related RFC threads: Issue #830 (TemporalParquet spec), PR #911 (TemporalParquet doc), Discussion #861 (portable SQL dialect).

Test suite: `make test` — 952 assertions across 36 test files.
